h2o-core/src/main/java/water/rapids/Merge.java

```diff
@@ -327,26 +329,145 @@ public void map(Chunk[] cs, NewChunk[] ncs) {
     }
     Key<Vec> key = Vec.newKey();
     Vec[] vecs = new Vec(key, Vec.ESPC.rowLayout(key, espc)).makeCons(numColsInResult, 0, doms, types);
-    System.out.println("took: " + (System.nanoTime() - t0) / 1e9);
+    Log.debug("took: " + (System.nanoTime() - t0) / 1e9+" seconds.");
 
-    System.out.print("Finally stitch together by overwriting dummies ...");
+    Log.info("Finally stitch together by overwriting dummies ...");
     t0 = System.nanoTime();
     Frame fr = new Frame(names, vecs);
     ChunkStitcher ff = new ChunkStitcher(chunkSizes, chunkLeftMSB, chunkRightMSB, chunkBatch);
     ff.doAll(fr);
-    System.out.println("took: " + (System.nanoTime() - t0) / 1e9);
+    Log.debug("took: " + (System.nanoTime() - t0) / 1e9+" seconds");
 
     //Merge.cleanUp();
     return fr;
   }
 
-  private static RadixOrder createIndex(boolean isLeft, Frame fr, int[] cols, int[][] id_maps, int[] ascending) {
-    System.out.println("\nCreating "+(isLeft ? "left" : "right")+" index ...");
+  public static ArrayList<SortCombine> gatherSameMSBRows(Frame leftFrame) {
+    long t0 = System.nanoTime();
+    ArrayList<SortCombine> bmList = new ArrayList<>();
```

Comment:
knitpicking: but you should use interfaces for collections when possible (e.g. List<SortCombine> bmList here), there are few more cases of this in this PR

Reply:
I'd personally stress this more than the formatting. Let's use the `List` interface there please.